lib/codegen/Decision.cpp

```diff
@@ -1070,6 +1070,7 @@ void makeStepFunction(
   codegen(dt);
 }
 
+// TODO: As we're not using fastcc, we must check if we can remove this wrapper
```

Comment:
```suggestion
```

I suspect that the same issues you experienced with `fastcc` will occur again when using `tailcc`. The LLVM [language specification](https://llvm.org/docs/LangRef.html#calling-conventions) states that `tailcc` is equivalent to `fastcc` except with the addition of guaranteed tail calls. The ABI changes that break GDB et. al. will therefore still be present with your changes here.